services/rsources/handler.go

```diff
@@ -5,13 +5,26 @@ import (
 	"database/sql"
 	"encoding/json"
 	"fmt"
+	"regexp"
+	"strings"
+	"time"
 
 	"github.com/lib/pq"
-	"github.com/rudderlabs/rudder-server/services/rsources/internal/extension"
+	"github.com/rudderlabs/rudder-server/config"
 )
 
+// In postgres, the replication slot name can contain lower-case letters, underscore characters, and numbers.
+var replSlotDisallowedChars *regexp.Regexp
+
+func init() {
+	replSlotDisallowedChars = regexp.MustCompile(`[^a-z0-9_]`)
+}
```

Comment:
[question] Just curious, is `init()` really needed here? Why not just do 
```go
var replSlotDisallowedChars = regexp.MustCompile(`[^a-z0-9_]`)
```

Reply:
I think I have mention init in the previous comment, but what you describe works as well (and is simpler)